gax/src/main/java/com/google/api/gax/batching/v2/BatchingDescriptor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.batching.v2;
+
+import com.google.api.core.SettableApiFuture;
+import java.util.List;
+
+/**
+ * This represents operations that transform elements/result data for batching purposes.
+ *
+ * <p>Implementations of BatchingDescriptor must guarantee that all methods are stateless and thread
+ * safe. It will be autogenerated by the gapic-generator.
+ */
```

Comment:
An adapter that packs and unpacks the elements in and out of batch requests and responses. This interface should be implemented by either a service specific client or auto generated by gapic-generator. 

Example implementation:
```code
class ListBatchingDescriptor<String,String,List<String>,List<String> {
  RequestBuilder<String, List<String>> newRequestBuilder(String prototype) {
    return new RequestBuilder<String, List<String>> {
      List<String> list = prototype.clone();
      void add(String element) {
        list.add(element);
      }
      List<String> build() {
        return list.clone();
      }
    }
  }
  void splitResponse(List<String> batchResponse, List<SettableApiFuture<String>> batch) {
     for(int i=0; i<batchResponse.size(); i++) {
       batch.get(i).set(batchResponse.get(i);
     }
   }
 ....
}
```